admin/controllers/pupil.js

```diff
@@ -187,39 +185,38 @@ controller.postEditPupil = async (req, res, next) => {
 
   const trimAndUppercase = R.compose(R.toUpper, R.trim)
 
-  pupil._id = req.body._id
-  pupil.foreName = req.body.foreName
-  pupil.middleNames = req.body.middleNames
-  pupil.lastName = req.body.lastName
-  pupil.upn = trimAndUppercase(R.pathOr('', ['body', 'upn'], req))
-  pupil.gender = req.body.gender
-  pupil.pin = pupil.pin || null
-  pupil.pinExpired = pupil.pinExpired || false
-  pupil.dob = dateService.createFromDayMonthYear(req.body['dob-day'], req.body['dob-month'], req.body['dob-year'])
-
+  const update = {
+    id: pupil.id,
+    foreName: req.body.foreName,
+    middleNames: req.body.middleNames,
+    lastName: req.body.lastName,
+    upn: trimAndUppercase(R.pathOr('', ['body', 'upn'], req)),
+    gender: req.body.gender,
+    dateOfBirth: dateService.createFromDayMonthYear(req.body['dob-day'], req.body['dob-month'], req.body['dob-year'])
+  }
   try {
-    await pupilDataService.update({_id: pupil._id}, pupil)
+    await pupilDataService.sqlUpdate(update)
     req.flash('info', 'Changes to pupil details have been saved')
   } catch (error) {
-    // TODO: handle mongoose validation error?
     next(error)
   }
 
   // pupil saved - redirect and highlight the saved pupil
-  const pupilId = JSON.stringify([pupil._id])
-  res.redirect(`/school/pupil-register/lastName/true?hl=${pupilId}`)
+  const highlight = JSON.stringify([pupil.urlSlug.toString()])
+  res.redirect(`/school/pupil-register/lastName/true?hl=${highlight}`)
 }
 
 controller.getPrintPupils = async (req, res, next) => {
   res.locals.pageTitle = 'Print pupils'
   let pupilsFormatted
   try {
-    const {pupils, schoolData} = await pupilDataService.getPupils(req.user.School)
-    const pupilsData = pupils.map(e => e.toJSON()).filter(p => !!p.pin && !p.pinExpired)
+    const pupils = await pupilDataService.sqlFindPupilsByDfeNumber(req.user.School)
+    const pupilsData = pupils.filter(p => !!p.pin)
```

Comment:
how come you changed the filter?
it used to be `.filter(p => !!p.pin && !p.pinExpired)`

Reply:
pupil.pinExpired is not present in the new sql schema.  @GuyHarwood  - care to comment on why it was removed?

Reply:
@GeorgeChatzigiannis - in fact looking at the mongo model `pinExpired` is not on that either.  How did it ever get set?

There is a field called `pinExpiresAt` which takes a datetime.

Reply:
pinExpired was superfluous, as it can be easily inferred from the pinExpiresAt field

Reply:
so the filter should change to take that into account then?

Reply:
Okay - I'll update the filter using pinExpired